mmv1/third_party/terraform/tests/resource_compute_region_network_endpoint_group_test.go.erb

```diff
@@ -0,0 +1,78 @@
+<% autogen_exception -%>
+package google
+
+<% unless version == 'ga' -%>
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+<% end -%>
+
+<% unless version == 'ga' -%>
+func TestAccComputeRegionNetworkEndpointGroup_negWithServerlessDeployment(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeRegionNetworkEndpointGroupDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeRegionNetworkEndpointGroup_negWithServerlessDeployment(context),
+			},
+			{
+				ResourceName:            "google_compute_region_network_endpoint_group.test_neg",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"region"},
+			},
+		},
+	})
+}
+<% end -%>
+
+<% unless version == 'ga' -%>
+func testAccComputeRegionNetworkEndpointGroup_negWithServerlessDeployment(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_api_gateway_api" "api_gw" {
+	api_id = "tf-test-%{random_suffix}"
+}
+		
+resource "google_api_gateway_api_config" "api_gw" {
+	api = google_api_gateway_api.api_gw.api_id
+	api_config_id = "tf-test-config-%{random_suffix}"
+
+	openapi_documents {
+		document {
+			path = "spec.yaml"
+			contents = filebase64("test-fixtures/apigateway/openapi.yaml")
+		}
+	}
+
+	lifecycle {
+		create_before_destroy = true
+	}
+}
+
+resource "google_api_gateway_gateway" "api_gw" {
+	api_config = google_api_gateway_api_config.api_gw.id
+	gateway_id = "tf-test-%{random_suffix}"
+}
+
+resource "google_compute_region_network_endpoint_group" "test_neg" {
+	name                  = "test-serverless-neg"
+	network_endpoint_type = "SERVERLESS"
+	region                = "us-central1"
+	serverless_deployment {
+		platform = "apigateway.googleapis.com"
+		url_mask = format("<gateway>%s/hello", trimprefix(google_api_gateway_gateway.api_gw.default_hostname, "tf-test-%{random_suffix}"))
```

Comment:
I couldn't find this thing documented anywhere, and it was quite difficult for me to extract from the docs, both Google GCP and Terraform, what value needs to be put in here for the serverless API Gateway deployment.

Reply:
faced the same thing. I think it should be documented somewhere